compiler/src/iree/compiler/API/Internal/Embed.cpp

```diff
@@ -73,11 +64,20 @@
 #include "mlir/Parser/Parser.h"
 #include "mlir/Support/FileUtilities.h"
 #include "mlir/Support/LogicalResult.h"
+#include "llvm/Support/Allocator.h"
```

Comment:
Ew does LLVM style hardcode that LLVM includes always go last?

Reply:
It's possible to disable this if we wanted to with [`IncludeCategories`](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#includecategories).

An example: https://github.com/GPUOpen-Drivers/llpc/blob/5ba112ea7a0a394ec2c25d394380bfb33af77600/.clang-format#L5

Reply:
I'm kind of inclined to do that. That doesn't affect code snippets between projects and if something is getting moved it probably needs its includes tweaked anyway. It seems like something that projects commonly modify. Is there a way to actually *view* what a `BasedOnStyle` does?

Reply:
What ordering would we prefer? The existing one or something close to that?

> Is there a way to actually view what a BasedOnStyle does?

Yes, `--dump-config` does that. I uploaded both dumps here: https://gist.github.com/kuhar/c8aa91e494b22fc7a60355fdbdf6ad70.

Reply:
IDK I'm probably not the right one to decide that. The Google one looks nice and generic just putting system headers and the "main" header first. The LLVM one looks pretty project-specific.

Reply:
I switched back to the previous include ordering as it also looked odd to me to be making a special case for llvm headers. This does not compromise on the goal of easier code sharing/movement, and reduced the amount of changes overall.

Reply:
I think the rationale is that includes are sorted from more specific to more generic to avoid missing dependencies. For example:
```
#include <system>
#include "llvm.h"
```
This would compile even if "llvm.h" needs `<system>` but it's accidentally not included in "llvm.h". Given that most of the project depends on llvm and not the other way around, I guess it makes sense to move llvm include right before system. I'm just guessing here, though :)

Reply:
https://llvm.org/docs/CodingStandards.html#include-style has more info